File: TesteMongeral/app/src/main/java/testemongeral/com/br/testemongeral/activity/MainActivity.java
package testemongeral.com.br.testemongeral.activity;

import android.app.FragmentManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import testemongeral.com.br.testemongeral.R;
import testemongeral.com.br.testemongeral.fragment.ChangeFragment;
import testemongeral.com.br.testemongeral.fragment.RepositoryFragment;

public class MainActivity extends AppCompatActivity {

    Toolbar myToolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        myToolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(myToolbar);
        ChangeFragment.changeFragment(R.id.conatiner, RepositoryFragment.class, getFragmentManager(), false, null);
   }

    public Toolbar getMyToolbar() {
        return myToolbar;
    }

}

File: desafioandroid/src/main/java/com/renanparis/desafioandroid/ui/activity/MainActivity.kt
package com.renanparis.desafioandroid.ui.activity

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.navigation.NavController
import androidx.navigation.NavDestination
import androidx.navigation.Navigation
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupWithNavController
import com.renanparis.desafioandroid.R

class MainActivity : AppCompatActivity(), NavController.OnDestinationChangedListener {

    private lateinit var navController: NavController
    private lateinit var listToolbar: Toolbar
    private lateinit var detailToolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        navController = Navigation.findNavController(this, R.id.activity_main_nav_host)
        listToolbar = findViewById(R.id.toolbar_list)
        detailToolbar = findViewById(R.id.toolbar_detail)
        setupActionBar()
        navController.addOnDestinationChangedListener(this)
    }

    private fun setupActionBar() {
        val appBarConfig = AppBarConfiguration(setOf(R.id.productsListFragment, R.id.productDetailsFragment))

        if (listToolbar.visibility == View.VISIBLE) {
            setSupportActionBar(listToolbar)
            listToolbar.setupWithNavController(navController, appBarConfig)
        } else if (detailToolbar.visibility == View.VISIBLE) {
            setSupportActionBar(detailToolbar)
            detailToolbar.setupWithNavController(navController, appBarConfig)
        }
    }
    override fun onDestinationChanged(controller: NavController, destination: NavDestination, arguments: Bundle?) {
        if (destination.id == R.id.productDetailsFragment) {
            listToolbar.visibility = View.GONE
            detailToolbar.visibility = View.VISIBLE
            window.statusBarColor = resources.getColor(R.color.lightGray)
        } else {
            listToolbar.visibility = View.VISIBLE
            detailToolbar.visibility = View.GONE
            window.statusBarColor = resources.getColor(R.color.colorPrimaryDark)
        }
        setupActionBar()
    }
}
